### PR TITLE
BUG: raise KeyError if MultiIndex.get_loc_level is asked unused label

### LINKED ISSUE #22221: mi.drop(x).get_loc_level(x) returns empty slice (rather than raising KeyError)
#### Code Sample

```python
import pandas as pd

df = pd.DataFrame(dict(value=[0, 1], group=['filled','empty']))
groups = df.groupby('group')

def remove1 (group):
	return group[group.value != 1]['value']
trimmed = groups.apply(remove1)


print("Trimmed.loc['empty']:\n",trimmed.loc['empty'])
```
#### Problem description

Since Pandas 0.23, if a group (ex. 'empty') is emptied by GroupBy.apply(), then accessing the result (ex. trimmed) through trimmed.loc['empty'] returns the error ```KeyError: 'the label [empty] is not in the [index]'``` (full output below). This is possible related to https://github.com/pandas-dev/pandas/issues/21624.

Despite what the error says, printing trimmed.indexes shows that 'empty' is a valid index:
```
MultiIndex(levels=[['empty', 'filled'], [0, 2]],
           labels=[[1, 1], [0, 1]],
           names=['group', None])
```

Full Output:
```
Traceback (most recent call last):
  File ".../python3.6/site-packages/pandas/core/indexing.py", line 1790, in _validate_key
    error()
  File ".../python3.6/site-packages/pandas/core/indexing.py", line 1785, in error
    axis=self.obj._get_axis_name(axis)))
KeyError: 'the label [empty] is not in the [index]'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "pandasExample.py", line 14, in <module>
    print("Trimmed.loc['empty']:\n",trimmed.loc['empty'])
  File ".../python3.6/site-packages/pandas/core/indexing.py", line 1478, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File ".../python3.6/site-packages/pandas/core/indexing.py", line 1911, in _getitem_axis
    self._validate_key(key, axis)
  File ".../python3.6/site-packages/pandas/core/indexing.py", line 1798, in _validate_key
    error()
  File ".../python3.6/site-packages/pandas/core/indexing.py", line 1785, in error
    axis=self.obj._get_axis_name(axis)))
KeyError: 'the label [empty] is not in the [index]'
```
#### Expected Output

Prior to 0.23, an empty Series would be returned.

0.22.0 output:
```
Trimmed.loc['empty']:
 Series([], Name: value, dtype: int64)
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.0.final.0
python-bits: 64
OS: Darwin
OS-release: 17.4.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.3
pytest: 3.6.4
pip: 10.0.1
setuptools: 39.2.0
Cython: None
numpy: 1.15.0
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 6.5.0
sphinx: 1.7.6
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.1
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by toobaz**
- [x] closes #22221
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/multi/test_indexing.py [hunk 1]
```diff
@@ -271,6 +271,10 @@ def test_get_loc_level():
     assert new_index is None
 
     pytest.raises(KeyError, index.get_loc_level, (2, 2))
+    # GH 22221: unused label
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_indexing.py]: can you add the issue number here

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -2443,14 +2438,21 @@ def convert_indexer(start, stop, step, indexer=indexer, labels=labels):
...
-            loc = level_index.get_loc(key)
-            if isinstance(loc, slice):
-                return loc
-            elif level > 0 or self.lexsort_depth == 0:
-                return np.array(labels == loc, dtype=bool)
+            code = level_index.get_loc(key)
+
+            if level > 0 or self.lexsort_depth == 0:
+                # Desired level is not sorted
+                locs = np.array(labels == code, dtype=bool)
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: I think you can pass ``copy=False`` here

**Review by jreback — CHANGES_REQUESTED** [review]
small comments

**Comment by jreback** [comment]
can you also add the original OP test somewhere.

File: pandas/tests/indexes/multi/test_indexing.py (hunk 1 above)
Comment (toobaz) [reply pandas/tests/indexes/multi/test_indexing.py]: Do you mean repeat it below?

File: pandas/core/indexes/multi.py (hunk 2 above)
Comment (jreback) [reply pandas/core/indexes/multi.py]: can you do this

File: pandas/tests/indexes/multi/test_indexing.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/indexes/multi/test_indexing.py]: this is good as written

**Review by jreback — CHANGES_REQUESTED** [review]
minor comment

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @toobaz

**Pull request merged**